platform/jvm/core/src/ffi.rs

```diff
@@ -150,52 +187,46 @@ pub(crate) fn jobject_list_to_fields(
   Ok(fields)
 }
 
-/// Returns `AnnotatedLogFields` copied from the provided `JMap`.
-/// Internally does a lossy conversion into UTF-8 per [`to_string_lossy`](
-/// https://docs.rs/jni/latest/jni/strings/struct.JNIStr.html#method.to_string_lossy).
-pub fn jobject_map_to_fields(
+/// Converts parallel Java String arrays (keys and values) into `AnnotatedLogFields`.
+/// This is more efficient than using Field objects because it avoids:
+/// 1. Creating Field wrapper objects on the Kotlin side
+/// 2. Multiple JNI method calls per field (getKey, getValueType, getValue)
+///
+/// The keys and values arrays must have the same length - keys[i] corresponds to values[i].
+pub fn string_arrays_to_annotated_fields(
```

Comment:
This used to support passing binary field values is that no longer required? Maybe it's handled elsewhere?

I haven't read the rest of the PR but how come we can't consolidate the logic to using one way to convert JVM values to log fields? Seems like we can figure out the best way and just use that everywhere?